docs/developing/migrating.md

```diff
@@ -1 +1,116 @@
-# Migrating your Java Code into Kotlin
\ No newline at end of file
+# Migrating your Java Code into Kotlin
+
+A good way to learn how Kotlin works is to convert your existing Java code into Kotlin.
+Kotlin is interoperable with Java, so you can use both languages in the same project.
+
+## Adding Kotlin to your Project
+
+### Adding the Kotlin Plugin
+
+To enable Kotlin in your WPILib project, add the Kotlin plugin to your Gradle build file.
+In the `build.gradle` file, add the following plugin to the `plugins` block:
+
+```gradle
+plugins {
+    id 'org.jetbrains.kotlin.jvm' version '2.0.20'
+}
+```
+
+### Add Kotlin Usage Reporting
+
+To help the FRC community keep track of how many teams use Kotlin as a programming language, we recommend adding the
+following lines to your `Robot.java` constructor
+
+```java
+HAL.report(tResourceType.kResourceType_Language, tInstances.kLanguage_Kotlin);
+```
+
+## Creating a new Kotlin File
+
+To create a new Kotlin file, right-click on the `src/main/java` folder in your project and select
```

Comment:
Do we wanna recommend src/main/kotlin or something similar?

Reply:
Eh idk I've always been on the fence about it, since some teams still use a bit of java code and then you're either still putting code in a directory that doesnt make sense, or you have two code directories which just feels bloated to me

Reply:
Yeah I think leaving as src/main/java, especially since the FRC plugin creates kotlin projects with `src/main/java`